From katzj@redhat.com  Mon Jun 25 15:09:15 2007
From: Jeremy Katz <katzj@redhat.com>
Date: Tue, 19 Jun 2007 17:16:10 -0400
Subject: USB: Don't autosuspend Blackberry devices.
To: gregkh@suse.de
Cc: linux-usb-devel@lists.sourceforge.net, Jeremy Katz <katzj@redhat.com>
Message-ID: <11822877702060-git-send-email-katzj@redhat.com>


Blackberry devices charge over USB.  By autosuspending the port, they are
not able to charge reliably.

Signed-off-by: Jeremy Katz <katzj@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/quirks.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -38,6 +38,11 @@ static const struct usb_device_id usb_qu
 	/* Philips PSC805 audio device */
 	{ USB_DEVICE(0x0471, 0x0155), .driver_info = USB_QUIRK_RESET_RESUME },
 
+	/* RIM Blackberry */
+	{ USB_DEVICE(0x0fca, 0x0001), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+	{ USB_DEVICE(0x0fca, 0x0004), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+	{ USB_DEVICE(0x0fca, 0x0006), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+
 	{ }  /* terminating entry must be last */
 };
 
